internal/auth0/organization/data_source.go

```diff
@@ -0,0 +1,166 @@
+		"If not provided, `organization_id` must be set. " +
+		"For performance, it is advised to use the `organization_id` as a lookup if possible."
+	dataSourceSchema["name"].AtLeastOneOf = []string{"organization_id", "name"}
+
+	dataSourceSchema["connections"] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"connection_id": {
+					Type:        schema.TypeString,
+					Computed:    true,
+					Description: "The ID of the enabled connection on the organization.",
+				},
+				"assign_membership_on_login": {
+					Type:     schema.TypeBool,
+					Computed: true,
+					Description: "When `true`, all users that log in with this connection will be " +
+						"automatically granted membership in the organization. When `false`, users must be " +
+						"granted membership in the organization before logging in with this connection.",
+				},
+			},
+		},
+	}
+
+	return dataSourceSchema
+}
+
+func readOrganizationForDataSource(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+	var foundOrganization *management.Organization
+	var err error
+
+	organizationID := data.Get("organization_id").(string)
+	if organizationID != "" {
+		foundOrganization, err = api.Organization.Read(organizationID)
+		if err != nil {
+			if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+				data.SetId("")
+				return nil
+			}
+			return diag.FromErr(err)
+		}
+	} else {
+		name := data.Get("name").(string)
+		page := 0
+
+	outerLoop:
+		for {
+			organizations, err := api.Organization.List(management.Page(page), management.PerPage(100))
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			for _, organization := range organizations.Organizations {
+				if organization.GetName() == name {
+					foundOrganization = organization
+					break outerLoop
+				}
+			}
+
+			if !organizations.HasNext() {
+				break
+			}
+
+			page++
+		}
+
+		if foundOrganization == nil {
+			return diag.Errorf("No organization found with \"name\" = %q", name)
+		}
+	}
+
+	data.SetId(foundOrganization.GetID())
+
+	result := multierror.Append(
+		data.Set("name", foundOrganization.GetName()),
+		data.Set("display_name", foundOrganization.GetDisplayName()),
+		data.Set("branding", flattenOrganizationBranding(foundOrganization.GetBranding())),
+		data.Set("metadata", foundOrganization.GetMetadata()),
+	)
+
+	foundConnections, err := fetchAllOrganizationConnections(api, foundOrganization.GetID())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	result = multierror.Append(
+		result,
+		data.Set("connections", flattenOrganizationConnections(foundConnections)),
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func fetchAllOrganizationConnections(api *management.Management, organizationID string) ([]*management.OrganizationConnection, error) {
```

Comment:
In the future we can consider using a generic helper for the paging logic, which seems to be repeated in [a number of places](https://github.com/search?q=repo%3Aauth0%2Fterraform-provider-auth0%20HasNext&type=code).

Reply:
IMO a better place for this would be the Go SDK itself. It would be nice if it would provide functions like "ListAll()" to wrap that logic.

Reply:
Yes, good idea!